state-network.md

```diff
@@ -255,53 +269,297 @@ sedes      := Container(enrs: List[byte_list, max_length=32], payload: byte_list
+                /                         \
+1:             0                           1
+             /   \
+           /       \
+         /           \
+2:      0             1
+       / \
+      /   \
+3:   0     1
+          / \
+4:       0  (1)
+
+```
+
+Note that it also includes the intermediate trie nodes under the paths:
+
+- `000`
+- `001`
+- `00`
+- `01`
+- `0`
+- `1`
+- `X`
+
+The gossip payload for the trie node at `0011` will contain these trie nodes as a proof.  Upon receipt of this trie node and proof, the receiving DHT node will extract the proof for the intermediate node under the path `001` which is the direct parent of the main trie node currently being gossiped.  This proof would be visualized as follows:
+
+
+```
+0:                           X
+                            / \
+                          /     \
+                        /         \
+                      /             \
+                    /                 \
+                  /                     \
+                /                         \
+1:             0                           1
+             /   \
+           /       \
+         /           \
+2:      0             1
+       / \
+      /   \
+3:   0    (1)
+
+4:
+
+```
+
+The DHT node would then *initialize* gossip for the trie node under the path `001`, using this new slimmed down proof to anchor to the same state root.  This process continues until it reaches the state root where it naturally terminates.  We refer to this as *recursive trie node* gossip.
+
+#### Stage 2C: Initialization Of Leaf Proof Gossip
+
+When receiving an individual trie node which represents a "leaf" node in the trie, the combined leaf node and proof are equivalent to a *"leaf proof"*.  The DHT node receiving this data in the form of "trie node" data is responsible for *initializing gossip* for the same data as a *"leaf proof"*.
+
+#### Stage 3: Neighborhood *"Leaf Proof"* Gossip
+
+When receiving a *"leaf proof"* over gossip, a DHT node will perform *"neighborhood gossip"* to nearby nodes from their routing table.
+
+
+### Updating cold Leaf Proofs
+
+Anytime the state root changes for either the main account trie or a contract storage trie, every leaf proof under that root will need to be updated.  The primary gossip mechanism will ensure that leaf data that was added, modified, or removed will receive and updated proof.  However, we need a mechanism for updating the leaf proofs for "cold" data that has not been changed.
+
+Each time a new block is added to the chain, the DHT nodes storing leaf proof data will need to perform a walk of the trie starting at the state root.  This walk of the trie will be directed towards the slice of the trie dictated by the set of leaves that the node is storing.  As the trie is walked it should be compared to the previous proof from the previous state root.  This walk concludes once the node has fully connected the leaf data from there previous proof to the new state root.
```

Comment:
I'm not sure but I think I only understood this because I knew what you were going to say; I don't think it's obvious what "fully connected" means. Maybe something like: "This walk concludes once it reaches a part of the tree which has not changed under the new state root, meaning the node has all the information it needs to be able to prove the cold leaves are still valid under the new state root." I don't know, maybe another picture would be better, showing that it's just the top of the proof which needs to be updated.

Reply:
An option:
```suggestion
Each time a new block is added to the chain, the DHT nodes storing leaf proof data will need to perform a walk of the trie starting at the state root.  This walk of the trie will be directed towards the slice of the trie dictated by the set of leaves that the node is storing.  As the trie is walked it should be compared to the previous proof from the previous state root.  This walk concludes once all of the in-range leaves can be proven with the new state root.
```